client/js/util/touch.js

```diff
@@ -0,0 +1,85 @@
+const direction = {
+    NONE: null,
+    LEFT: 'left',
+    RIGHT: 'right',
+    DOWN: 'down',
+    UP: 'up'
+};
+
+function handleTouchStart(handler, evt) {
+    const touchEvent = evt.touches[0];
+    handler._xStart = touchEvent.clientX;
+    handler._yStart = touchEvent.clientY;
+}
+
+function handleTouchMove(handler, evt) {
+    if (!handler._xStart || !handler._yStart) {
+        return;
+    }
+
+    const xDirection = handler._xStart - evt.touches[0].clientX;
+    const yDirection = handler._yStart - evt.touches[0].clientY;
+
+    if (Math.abs(xDirection) > Math.abs(yDirection)) {
+        if (xDirection > 0) {
+            handler._direction = direction.LEFT;
+        } else {
+            handler._direction = direction.RIGHT;
+        }
+    } else {
+        if (yDirection > 0) {
+            handler._direction = direction.DOWN;
+        } else {
+            handler._direction = direction.UP;
```

Comment:
When testing this functionality, I found that up and down are actually reversed here.